models/DataObject/ClassDefinition/Data/Extension/Text.php

```diff
@@ -36,7 +36,7 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
      */
     public function isEmpty($data)
     {
-        return strlen($data) < 1;
+        return empty($data) || strlen($data) < 1;
```

Comment:
I'm not entirely sure where this is used, but `empty` is dangerous, as it the string `"0"` is considered empty, too (see: https://3v4l.org/mgSdd).

Reply:
Right, which is why I don't use it, but was suggested by @brusch in the other PR. :) I guess this needs some more thinking.

Reply:
Well, `$data` is expected to be `string` or `null`, so why not just cast it to `string` then? => https://3v4l.org/57cNL

Reply:
I try to avoid type casts because they're quite fragile. It's better to handle the intended types directly in the code IMO. If you know null is possible here, check for null. If others things start creeping in, that will allow it to fail, but casting will silently "help".

Reply:
$data should be string or null. So this should be enough:
```suggestion
        return $data === null || strlen($data) < 1;
```

Reply:
Using `!is_string($data)` is dangerous here, as it is a possible BC break - see failing tests.
Reason: currently it also works with numeric data such as `2345` which is not a `string` and would treat it as empty.